pkg/diagnostic/kms/check.go

```diff
@@ -23,46 +28,130 @@ var (
 )
 
 type checker struct {
-	cfg    *config.Settings
-	logger *logrus.Entry
+	cfg       *config.Settings
+	logger    *logrus.Entry
+	clientset kubernetes.Interface
 }
 
-func NewProvider(ctx context.Context, cfg *config.Settings) diagnostic.Provider {
+var NewProvider = func(ctx context.Context, cfg *config.Settings, clientset ...kubernetes.Interface) diagnostic.Provider {
+	var cs kubernetes.Interface
+	if len(clientset) > 0 {
```

Comment:
just curious, why do we need this?

Reply:
`clientset` is an optional arg, mainly used for unit tests. This is just a way of checking that one is being passed and if so, prefer that over finding it dynamically based on the `InClusterConfig`. For a bit of background, mocking the `InClusterConfig` proved to be difficult, so this was an easy alternative.